ocaml/testsuite/tests/typing-modules/package_constraint.ml

```diff
@@ -0,0 +1,332 @@
+(* TEST
```

Comment:
This file is different than the version upstream. Maybe separate out our local changes (just by putting them at the bottom) to make merging (and reviewing) easier?

Reply:
Done (well, mostly - one addition makes a lot of sense where it is, and I think doesn't meaningfuly impact future merges becuase of how many nearby errors also change in our version).